### PR TITLE
Update to podman 5, including netavark and pasta

### PR DESCRIPTION
**Pull request by mattcen**
- Remove slirp4netns, and replace with pasta
- Build and install netavark
- Upgrade to podman 5.0.0

### CODE REVIEW

**Comment by mgoltzsche** [comment]
Hi @mattcen, thanks for creating the PR!
Though, the CI pipeline fails because of the following issues:
* The podman remote tests fail because the podman [version within `Dockerfile-remote`](https://github.com/mgoltzsche/podman-static/blob/v4.9.4/Dockerfile-remote#L12) needs to be updated correspondingly.
* The podman networking tests fail because the `netavark` binary  requires libgcc to be linked dynamically, e.g. see:
```
not ok 3 rootful podman - internet connectivity (CNI networking) in 1738ms
# (in test file test/rootful.bats, line 18)
#   `$DOCKER run --rm --privileged --entrypoint /bin/sh -u root:root \' failed with status 126
# Resolving "alpine" using unqualified-search registries (/etc/containers/registries.conf)
# Trying to pull docker.io/library/alpine:3.17...
# Getting image source signatures
# Copying blob sha256:3c854c8cbf469fda815b8f6183300c07cfa2fbb5703859ca79aff93ae934961b
# Copying config sha256:eaba187917ccc6af1386ecfdd95a56858a553d14888387980fb07b326e8ce890
# Writing manifest to image destination
# Error loading shared library libgcc_s.so.1: No such file or directory (needed by /usr/local/lib/podman/netavark)
# Error relocating /usr/local/lib/podman/netavark: _Unwind_Resume: symbol not found
# Error relocating /usr/local/lib/podman/netavark: _Unwind_GetIPInfo: symbol not found
# Error relocating /usr/local/lib/podman/netavark: _Unwind_GetDataRelBase: symbol not found
# Error relocating /usr/local/lib/podman/netavark: _Unwind_GetRegionStart: symbol not found
# Error relocating /usr/local/lib/podman/netavark: _Unwind_SetGR: symbol not found
# Error relocating /usr/local/lib/podman/netavark: _Unwind_GetTextRelBase: symbol not found
# Error relocating /usr/local/lib/podman/netavark: _Unwind_DeleteException: symbol not found
# Error relocating /usr/local/lib/podman/netavark: _Unwind_RaiseException: symbol not found
# Error relocating /usr/local/lib/podman/netavark: _Unwind_GetIP: symbol not found
# Error relocating /usr/local/lib/podman/netavark: _Unwind_Backtrace: symbol not found
# Error relocating /usr/local/lib/podman/netavark: _Unwind_GetLanguageSpecificData: symbol not found
# Error relocating /usr/local/lib/podman/netavark: _Unwind_SetIP: symbol not found
# Error: netavark: : EOF
```
Can we get rid of that dependency somehow?
Otherwise, can we statically link it?

**Comment by mattcen** [comment]
Heya!

I created this PR just because I wanted to to upgrade to pasta and decided to share my work. I'm not sure how much capacity I'll have to fix these issues in the immediate future, but I'll try to book some time to look into them.

Thanks for the feedback!

**Comment by mattcen** [comment]
I've pushed a fix to Dockerfile-remote. Apparently the tests all fail when I run them on my Mac, but I suspect that's because they make some Linux-specific assumptions.
Haven't dug into the netavark thing yet. Another thing to note on that though is that it might be worth building it from a different (Rust-specific) container base rather than waste time installing a bunch of Rust stuff into the Go Alpine image.

File: Makefile [hunk 1]
```diff
@@ -104,8 +104,7 @@ tar: .podman-from-container
 	cp -r conf/containers $(ASSET_DIR)/etc/containers
 	cp -r conf/cni $(ASSET_DIR)/etc/cni
 	cp README.md $(ASSET_DIR)/
-	cp -r $(IMAGE_ROOTFS)/usr/local/lib $(ASSET_DIR)/usr/local/lib
-	cp -r $(IMAGE_ROOTFS)/usr/local/bin $(ASSET_DIR)/usr/local/bin
+	$(DOCKER) run --rm $(PODMAN_IMAGE) tar c /usr/local/{bin,lib} | tar -xC $(ASSET_DIR)
```
Comment (mgoltzsche) [inline Makefile:107]: What's the purpose of this change?

File: Dockerfile [hunk 2]
```diff
@@ -182,6 +205,7 @@ COPY conf/crun-containers.conf /etc/containers/containers.conf
 # Build podman image with rootless binaries and CNI plugins
 FROM rootlesspodmanrunc AS podmanall
 RUN apk add --no-cache iptables ip6tables
+COPY --from=slirp4netns /slirp4netns/slirp4netns /usr/local/bin/slirp4netns
```
Comment (mgoltzsche) [inline Dockerfile:208]: Why did you add slirp4netns back? Did it turn out to be required still?

**Review by mgoltzsche — COMMENTED** [review]
Now that you also updated the podman version within the remote Dockerfile, the podman remote tests are passing.
However, the `netavark` binary is still dynamically linked to libgcc which makes the networking tests fail.
Given that the tests are not running on your machine, you could quickly verify that the `netavark` build is producing a statically linked binary by running `! ldd /usr/local/lib/podman/netavark` within the podman container. The command should terminate successfully or list the (missing) dynamically linked library. You could as well add that command to the Dockerfile step that builds the binary to make sure it is really statically linked.

File: Makefile (hunk 1 above)
Comment (mattcen) [reply Makefile:107]: Oh sorry, I pushed more commits for this PR than I intended.
I don't *fully* understand how the Makefile is exporting the images, but when I try it on my Mac, it doesn't behave as expected and I couldn't pull the contents from those dirs. I don't recall what the error was at this stage.

In any case, it's not a change that needs to be upstreamed; I'll pick out this commit when I get a chance.

Comment (mattcen) [reply Makefile:107]: Oh! I *think* the problem is that `.podman-from-container` would need to depend on `multiarch-images` for this to work, and because I wasn't *running* `multiarch-images`, which I didn't need, the images weren't being exported, thus the files being copied did not exist.

Does that sound like a viable explanation?

Comment (mgoltzsche) [reply Makefile:107]: I see. The purpose of this Makefile target is only to extract the binaries from the image in order to publish them as tar archive that is attached to each GitHub release. Thus, so far it was always run after the `multiarch-images` target in practice.

**Comment by mgoltzsche** [comment]
Thanks again @mattcen! I'll carry it from here...
